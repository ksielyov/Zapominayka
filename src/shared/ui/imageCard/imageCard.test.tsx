import React from 'react';
import renderer from 'react-test-renderer';
import {ImageCard} from '@ui';

test('imageCard should render correctly', () => {
  const tree = renderer
    .create(
      <ImageCard
        index={2}
        imageSource={
          'https://priioajn.files.wordpress.com/2019/08/landscape-2459981.jpg?w=1200'
        }
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
