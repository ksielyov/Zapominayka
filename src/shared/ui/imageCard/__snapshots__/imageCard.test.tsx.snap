// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imageCard should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#424242",
      "borderRadius": 23.541176470588233,
      "height": 307.9970588235294,
      "marginBottom": undefined,
      "opacity": 1,
      "width": 577.364705882353,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.69)",
        "borderBottomRightRadius": 23.541176470588233,
        "borderTopLeftRadius": 23.541176470588233,
        "height": 62.77647058823529,
        "justifyContent": "center",
        "position": "absolute",
        "width": 62.77647058823529,
        "zIndex": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      2
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 23.541176470588233,
          "height": "100%",
          "width": "100%",
        },
      ]
    }
  >
    <FastImageView
      onFastImageLoadEnd={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://priioajn.files.wordpress.com/2019/08/landscape-2459981.jpg?w=1200",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 23.541176470588233,
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
        "zIndex": 4,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={153.0176470588235}
      bbWidth={62.77647058823529}
      fill="none"
      focusable={false}
      height={153.0176470588235}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 153,
            "width": 62,
          },
        ]
      }
      vbHeight={156}
      vbWidth={72}
      width={62.77647058823529}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M27.49.016C19.882-.71-3.003 90.549 1.582 94.289c4.585 3.74 17.042-30.392 21.203-28.91 4.16 1.48 17.06 89.002 23.16 89.914 4.546.625 28.596-88.446 24.84-93.022-3.758-4.576-11.941 28.203-17.11 26.914C48.509 87.896 32.944.555 27.49.015z"
          fill={4280361249}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
